hardware_interface/include/adcs.h

```diff
@@ -0,0 +1,194 @@
+
+ADCS_returnState HAL_ADCS_get_measurements(adcs_measures* measurements);
+ADCS_returnState HAL_ADCS_get_actuator(adcs_actuator* commands);
+ADCS_returnState HAL_ADCS_get_estimation(adcs_estimate* data);
+ADCS_returnState HAL_ADCS_get_ASGP4(bool* complete, uint8_t* err,
+                                adcs_asgp4* asgp4);
+ADCS_returnState HAL_ADCS_get_raw_sensor(adcs_raw_sensor* measurements);
+ADCS_returnState HAL_ADCS_get_raw_GPS(adcs_raw_gps* measurements);
+ADCS_returnState HAL_ADCS_get_star_tracker(adcs_star_track* measurements);
+ADCS_returnState HAL_ADCS_get_MTM2_measurements(xyz16* Mag);
+ADCS_returnState HAL_ADCS_get_power_temp(adcs_pwr_temp* measurements);
+
+// ACP Config Msgs
+ADCS_returnState HAL_ADCS_set_power_control(uint8_t* control);
+ADCS_returnState HAL_ADCS_get_power_control(uint8_t* control);
+ADCS_returnState HAL_ADCS_set_attitude_angle(xyz att_angle);
+ADCS_returnState HAL_ADCS_get_attitude_angle(xyz* att_angle);
+ADCS_returnState HAL_ADCS_set_track_controller(xyz target);
+ADCS_returnState HAL_ADCS_get_track_controller(xyz* target);
+ADCS_returnState HAL_ADCS_set_log_config(uint8_t* flags_arr, uint16_t period,
+                                     uint8_t dest, uint8_t log);
+ADCS_returnState HAL_ADCS_get_log_config(uint8_t* flags_arr, uint16_t* period,
+                                     uint8_t* dest, uint8_t log);
+ADCS_returnState HAL_ADCS_set_inertial_ref(xyz iner_ref);
+ADCS_returnState HAL_ADCS_get_inertial_ref(xyz* iner_ref);
+
+ADCS_returnState HAL_ADCS_set_sgp4_orbit_params(adcs_sgp4 params);
+ADCS_returnState HAL_ADCS_get_sgp4_orbit_params(adcs_sgp4* params);
+ADCS_returnState HAL_ADCS_set_system_config(adcs_sysConfig config);
+ADCS_returnState HAL_ADCS_get_system_config(adcs_sysConfig* config);
+ADCS_returnState HAL_ADCS_set_MTQ_config(xyzu8 params);
+ADCS_returnState HAL_ADCS_set_RW_config(uint8_t* RW);
+ADCS_returnState HAL_ADCS_set_rate_gyro(rate_gyro_config params);
+ADCS_returnState HAL_ADCS_set_css_config(css_config config);
+ADCS_returnState HAL_ADCS_set_star_track_config(cubestar_config config);
+ADCS_returnState HAL_ADCS_set_cubesense_config(cubesense_config params);
+ADCS_returnState HAL_ADCS_set_mtm_config(mtm_config params, uint8_t mtm);
+ADCS_returnState HAL_ADCS_set_detumble_config(detumble_config config);
+ADCS_returnState HAL_ADCS_set_ywheel_config(ywheel_ctrl_config params);
+ADCS_returnState HAL_ADCS_set_rwheel_config(rwheel_ctrl_config params);
+ADCS_returnState HAL_ADCS_set_tracking_config(track_ctrl_config params);
+ADCS_returnState HAL_ADCS_set_MoI_mat(moment_inertia_config cell);
+ADCS_returnState HAL_ADCS_set_estimation_config(estimation_config config);
+ADCS_returnState HAL_ADCS_set_usercoded_setting(usercoded_setting setting);
+ADCS_returnState HAL_ADCS_set_asgp4_setting(aspg4_setting setting);
+ADCS_returnState HAL_ADCS_get_full_config(adcs_config* config);
+
+ADCS_returnState HAL_ADCS_getHK(ADCS_HouseKeeping* adcs_hk);
```

Comment:
Has this been implemented or is just the prototype made?

Reply:
See the .c file.

Reply:
yes, the actual implementation is in the adcs.c file